## ⚠️ Empirical finding: reviewer note was wrong on `cs/exposure-of-sensitive-information` location

First push (33a0827) trimmed the queryset to `security-extended` only, per the code-reviewer agent's nice-to-have note. The barrier-verification workflow then failed on this PR:

```
Probe alert counts — sanitized=0 unsanitized=0
##[error]Rule appears disabled — unsanitized probe raised 0 alerts.
```

This proves `cs/exposure-of-sensitive-information` is **not** included in `security-extended` for C# — it lives only in `security-and-quality`.

[Workflow run 25984484812](https://github.com/meepleAi-app/meepleai-monorepo/actions/runs/25984484812) is the receipt.

Revert pushed (a42cda5) restoring `security-and-quality`. The new file now includes a comment explicitly forbidding future re-trim attempts without prior empirical validation.

**Net change of this PR after revert**: only the workflow comment fix in `barrier-verification.yml` (CODEQL_ACTION_EXTRA_OPTIONS `format()` divergence rationale). The 1-2 min monthly-cron speedup hypothesis was incorrect.

This is actually a good outcome for the barrier-verification mechanism itself — it caught a regression in the same PR that introduced it. The "regression guard for the regression guard" worked as designed.

--- a/.github/workflows/barrier-verification.yml
+++ b/.github/workflows/barrier-verification.yml
@@ -19,14 +19,20 @@ on:
         required: false
         default: 'false'
         type: boolean
-  # Also run when the probes / config / workflow itself change so a PR can
-  # validate end-to-end before being merged.
-  pull_request:
-    paths:
-      - 'apps/api/tests/CodeqlBarrierSmoke/**'
-      - '.github/codeql/codeql-config-with-probes.yml'
-      - '.github/codeql/meepleai/sanitizers-extensions/**'
-      - '.github/workflows/barrier-verification.yml'
+  # NOTE — no `pull_request` trigger by design.
+  #
+  # Empirical finding on PR #1203 (2026-05-17): the codeql-action's PR-event
+  # default is incremental/diff-only analysis. The unsanitized probe alert
+  # is therefore filtered out on any PR that does not modify the probe
+  # source file itself (because it is "pre-existing" relative to the base
+  # branch). This makes a `pull_request` trigger here unreliable: it
+  # passes on PRs that touch the probe files and falsely fails on every
+  # other PR.
+  #
+  # Validation cadence is sufficient via the monthly cron + on-demand
+  # `workflow_dispatch`. PR authors who modify probes/config/MaD pack
+  # should run `gh workflow run barrier-verification.yml --field
+  # verbose=true --ref <branch>` to validate before merge.
 
 concurrency:
   group: barrier-verification-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}

